config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/AnnotationDeserializer.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2018 CA. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+/**
+ * AnnotationDeserializer deserializes the "annotations" tag in the entity elements. Annotation on entities are
+ * supported in 2 ways:
+ * 1) Array of objects containing "type" and the other optional fields like "name", "description" and "tags"
+ * field. For example,
+ * annotations:
+ * - type: "@bundle"
+ *   name: "encass-example"
+ *   description: "description for encass-example"
+ *   tags:
+ *   - "anytag"
+ *   - "sometag"
+ *
+ * 2) Array of Strings containing only "type". For example,
+ * annotations:
+ * - "@bundle"
+ * - "@reusable"
+ *
+ * This deserialization implementation takes care of both these types and deserializes the input into
+ * {@link java.util.Set} of {@link com.ca.apim.gateway.cagatewayconfig.beans.Annotation}
+ */
+package com.ca.apim.gateway.cagatewayconfig.bundle.builder;
+
+import com.ca.apim.gateway.cagatewayconfig.beans.Annotation;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.ObjectCodec;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import java.io.IOException;
+import java.util.*;
+
+public class AnnotationDeserializer extends JsonDeserializer<Set<Annotation>> {
```

Comment:
Like it the way you supported this mixture.

Reply:
thanks!